### ISSUE TITLE
Solidity programming language (*.sol) not recognised despite having filetype entry

### ISSUE THREAD

**Issue by PreciousChicken** [neovim/neovim#11666]
- `nvim --version`:

```bash
NVIM v0.4.3
Build type: RelWithDebInfo
LuaJIT 2.0.5
Compilation: /usr/bin/gcc-5 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -g -DMIN_LOG_LEVEL=3 -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wvla -fstack-protector-strong -fdiagnostics-color=auto -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/home/travis/build/neovim/bot-ci/build/neovim/build/config -I/home/travis/build/neovim/bot-ci/build/neovim/src -I/home/travis/build/neovim/bot-ci/build/neovim/.deps/usr/include -I/usr/include -I/home/travis/build/neovim/bot-ci/build/neovim/build/src/nvim/auto -I/home/travis/build/neovim/bot-ci/build/neovim/build/include
Compiled by travis@travis-job-9a287301-34f2-4b86-92e3-462594f51f36

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/home/travis/build/neovim/bot-ci/build/neovim/build/nvim.AppDir/usr/share/nvim
```

- Operating system/version:

```bash
OS: Ubuntu 18.04.3 LTS x86_64
Kernel: 4.15.0-72-generic
```

- Terminal name/version:
Suckless Simple Terminal (st)

- `$TERM`:
st-256color

### Steps to reproduce using `nvim -u NORC`
1.  Unzip attached file [example_sol.zip](https://github.com/neovim/neovim/files/4019847/example_sol.zip) (copied from [Solidity wikipedia entry](https://en.wikipedia.org/wiki/Solidity))
2. run `nvim -u NORC example.sol`
3. In normal mode type `:set filetype?`

### Actual behaviour

Returns `filetype=` (e.g. blank).

### Expected behaviour

Returns `filetype=solidity`

### Additional information

Typing in normal mode:

`:echo getcompletion('sol', 'filetype')`

returns `['solhint', 'solidity']`. 

Neovim should therefore recognise the language.  If it does not recognise the language then it should not be included in the list of recognised filetypes.

First contribution here, so entirely possible I am missing something or am just wrong...

**Comment by justinmk** [comment]
I see the same behavior in Vim 8.1.2200

> If it does not recognise the language then it should not be included in the list of recognised filetypes.

> `:echo getcompletion('sol', 'filetype')`
> 
> returns `['solhint', 'solidity']`.

For me it returns `[]`. Sounds like you have something in `pack/start` perhaps (assuming you really used `-u NORC`, check `:scriptnames`).

To get support for this file a ticket at https://github.com/vim/vim , we generally don't maintain  syntax files here.

**Comment by PreciousChicken** [comment]
Thank you for looking into this.

Following your response I booted up off a Kubuntu 18.04 LiveUSB and installed a fresh AppImage.  I was no longer able to replicate the error: like you I got a `[]`.  Clearly something in my configuration was amiss (still not sure what - there was nothing solidity related in `pack/start` at the time).  

Apologies for not having checked this out more thoroughly prior to posting.
